File: src/main/java/com/codingspezis/android/metalonly/player/plan/ShowInformation.java
package com.codingspezis.android.metalonly.player.plan;

import java.util.Date;

/**
 * Unification interface. Combines the common ops between both data classes. The operations are
 * explicitely named differently to enforce delegation to existing methods.
 */
public interface ShowInformation {
    String getModerator();
    String getGenre();
    String getShowTitle();
    Date getStartDate();
    Date getEndDate();
}

File: src/main/kotlin/com/codingspezis/android/metalonly/player/plan/ShowInformation.kt
package com.codingspezis.android.metalonly.player.plan

import java.util.Date

/**
 * Interface for ShowInformation.
 */
interface ShowInformation {
    val moderator: String
    val genre: String
    val showTitle: String
    val startDate: Date
    val endDate: Date
}
